src/test/mir-opt/const_prop/discriminant.rs

```diff
@@ -31,7 +31,7 @@ fn main() {
 // START rustc.main.ConstProp.after.mir
 //  bb0: {
 //      ...
-//      _3 = const {0x01: std::option::Option<bool>};
+//      _3 = const {transmute(0x1): std::option::Option<bool>};
```

Comment:
Is the `0x01` to `0x1` change intentional?

Reply:
Yes. The reason is https://github.com/rust-lang/rust/pull/67133#discussion_r374993484 though rereading that now makes me realize that @eddyb may have just meant to not print leading zeros on pointers?

Reply:
This is a loss of information, `0x01` vs `0x0001` encodes the bitwidth. I prefer the previous state.

For pointers leaving them away seems fine though as there is no ambiguity about their bitwidth.

Reply:
For pointers leaving them away seems fine though as there is no ambiguity about their bitwidth. I still have a slight preference to have leading zeroes everywhere, but if @eddyb feels strongly about this, so be it.

Reply:
> For pointers leaving them away seems fine though as there is no ambiguity about their bitwidth. I still have a slight preference to have leading zeroes everywhere

I don't mind this, I don't think. At the end of the day, it's very subjective, [and I have my own weird take](https://github.com/eddyb/sobek/blob/d32d777f8adca1411e606f9fa3b9da8c71e16dcb/src/ir/mod.rs#L57-L77) on it in my own project.

At the end of the day, if it works for testing and is ergonomic enough for users, it should be fine.

Reply:
Leading zeros are now only truncated for raw pointers, all other types show up with all leading zeros